lib/Onyx.d.ts

```diff
@@ -115,6 +120,21 @@ declare const METHOD: {
  */
 declare function getAllKeys(): Promise<Array<OnyxKey>>;
 
+/**
+ * Checks to see if the a subscriber's supplied key
+ * is associated with a collection of keys.
+ */
+declare function isCollectionKey(key: OnyxKey): key is CollectionKeyBase;
+
+declare function isCollectionMemberKey<TCollectionKey extends CollectionKeyBase>(collectionKey: TCollectionKey, key: string): key is `${TCollectionKey}${string}`;
+
+/**
+ * Splits a collection member key into the collection key part and the ID part.
+ * @param key - The collection member key to split.
+ * @returns A tuple where the first element is the collection part and the second element is the ID part.
+ */
+declare function splitCollectionMemberKey<TKey extends CollectionKey>(key: TKey): [TKey extends `${infer Prefix}_${string}` ? `${Prefix}_` : never, string];
```

Comment:
@blazejkustra Maybe you have some idea how to improve these typings? It should allow only collection member keys, but currently is allowing collection keys too.

```ts
Onyx.splitCollectionMemberKey(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`); // ["policy_", string], correct
Onyx.splitCollectionMemberKey(ONYXKEYS.COLLECTION.POLICY); // ["policy_", string], should raise an error in the param
```

Reply:
How about this?

```ts
type CollectionKeyBase = 'policy_' | 'report_';
type RemoveUnderscore<T> = T extends `${infer Word}_` ? Word : never;

const ONYXKEYS = {COLLECTION: {POLICY: 'policy_'}} as const;

declare function splitCollectionMemberKey<TKey extends RemoveUnderscore<CollectionKeyBase>, TRest extends string>(key: TRest extends '' ? never : `${TKey}_${TRest}`): [TKey, TRest];

splitCollectionMemberKey(`${ONYXKEYS.COLLECTION.POLICY}${123}`); // ["policy_", string], correct
splitCollectionMemberKey(ONYXKEYS.COLLECTION.POLICY);
```

Reply:
I'm not sure if it's worth it, as this type is quite complicated, up to you 👍

Reply:
Yeah, let's avoid overcomplicating and skip this for now.